topdown/cache/cache_test.go

```diff
@@ -139,6 +139,20 @@ func TestInsert(t *testing.T) {
 	if !found {
 		t.Fatal("Expected key \"foo5\" in cache")
 	}
+
+	// replacing an existing key should not affect cache size
+	cache = NewInterQueryCache(config)
+
+	cacheValue6 := newInterQueryCacheValue(ast.StringTerm("bar6").Value, 10)
+	cache.Insert(ast.StringTerm("foo6").Value, cacheValue6)
+	cache.Insert(ast.StringTerm("foo6").Value, cacheValue6)
+
+	cacheValue7 := newInterQueryCacheValue(ast.StringTerm("bar7").Value, 10)
+	dropped = cache.Insert(ast.StringTerm("foo7").Value, cacheValue7)
+
+	if dropped != 0 {
+		t.Fatal("Expected dropped to be zero")
+	}
```

Comment:
Can we add a test case like [this](https://github.com/open-policy-agent/opa/pull/5374/files#diff-51c6c5bd2ad2a3fe3f9a303771fa5c82ae818d15ed0bae1560384c8b88672e92R2564) to show the concurrent use case.